src/Mustache/Compiler.php

```diff
@@ -258,15 +262,42 @@ private function blockVar($nodes, $id, $start, $end, $otag, $ctag, $level)
     {
         $id = var_export($id, true);
 
-        return sprintf($this->prepare(self::BLOCK_VAR, $level), $id, $this->walk($nodes, 2));
+        return sprintf($this->prepare(self::BLOCK_VAR, $level), $id, $this->walk($nodes, $level));
     }
 
     const BLOCK_ARG = '
         // %s block_arg
-        $value = $this->section%s($context, \'\', true);
-        $newContext[%s] = %s$value;
+        $blockid = %s;
+        $newContext[%s] = array($this, \'block\' . $blockid);
```

Comment:
the $blockid is guaranteed to be a string (md5) so why not compile to a string literal? 

```php
$newContext[%s] = array($this, \'block%s\');
```

You'd need to pass in both the exported and unexported $id, but that seems worthwhile?